### PR TITLE
Tasks: shared bucket when in a group.

### PR DESCRIPTION
**Pull request by jortel**
Tasks within a group share the same bucket.

Also but unrelated, the Application facts should be initialized. This makes working with them so much easier.

### CODE REVIEW

File: api/taskgroup.go [hunk 1]
```diff
@@ -136,20 +136,27 @@ func (h TaskGroupHandler) Update(ctx *gin.Context) {
...
-	g.ID = id
+	g := &model.TaskGroup{}
+	result := h.DB.First(g, id)
+	if result.Error != nil {
+		h.getFailed(ctx, result.Error)
+		return
+	}
+	updated := r.Model()
+	updated.ID = g.ID
+	updated.Bucket = g.Bucket
```
Comment (jortel) [inline api/taskgroup.go:147]: Mainly need to fetch the group to ensure the bucket is propagated.  The caller should PUT a complete object fetched with GET but no guarantees and the bucket is too important.

File: api/application.go [hunk 2]
```diff
@@ -168,27 +167,27 @@ func (h ApplicationHandler) Delete(ctx *gin.Context) {
...
 	r := &Application{}
 	err := ctx.BindJSON(r)
 	if err != nil {
 		h.bindFailed(ctx, err)
 		return
 	}
 	m := r.Model()
-	nID, _ := strconv.Atoi(id)
-	m.ID = uint(nID)
-	result := h.DB.Model(&model.Application{}).Where("id = ?", id).Omit("id").Updates(m)
```
Comment (jortel) [inline api/application.go:181]: Mainly updated to test the Fact changes.

File: api/taskgroup.go [hunk 3]
```diff
@@ -136,21 +136,28 @@ func (h TaskGroupHandler) Update(ctx *gin.Context) {
...
+		h.updateFailed(ctx, result.Error)
+		return
+	}
+	updated := r.Model()
+	updated.ID = current.ID
+	updated.Bucket = current.Bucket
 	err = h.DB.Transaction(
 		func(tx *gorm.DB) (err error) {
-			db := tx.Model(g)
+			db := tx.Model(updated)
```
Comment (jortel) [inline api/taskgroup.go:150]: `g` renamed: `current` for clarity.

File: api/taskgroup.go [hunk 4]
```diff
@@ -136,21 +136,28 @@ func (h TaskGroupHandler) Update(ctx *gin.Context) {
...
-	g.ID = id
+	current := &model.TaskGroup{}
+	result := h.DB.First(current, id)
+	if result.Error != nil {
+		h.updateFailed(ctx, result.Error)
+		return
+	}
+	updated := r.Model()
+	updated.ID = current.ID
+	updated.Bucket = current.Bucket
```
Comment (jortel) [inline api/taskgroup.go:147]: Needed to propagate the group bucket to new tasks.

**Review by mansam — APPROVED** [review]

**Pull request merged**